client/src/modules/fiscal/templates/modals/fiscal.closing.modal.js

```diff
@@ -26,31 +65,35 @@ function ClosingFiscalYearModalController(Notify, Fiscal, Modal, Session, Instan
   .then(function (fiscal) {
     vm.fiscal = fiscal;
 
-    // get balance of period N of the year to close
+    // get balance until period N of the year to close
     return Fiscal.periodicBalance({
       id            : vm.fiscal.id,
       period_number : vm.fiscal.number_of_months, // last month
     });
   })
   .then(function (balance) {
-    var map = { income: 'profit', expense: 'charge' };
-
-    vm.exploitation = balance.reduce(function (previous, current) {
-      var key = map[current.type];
-
-      if (key) {
-        previous[key].push(current);
-      }
-
-      return previous;
-    }, { profit: [], charge: [] });
+    var profit = balance.filter(getProfitAccount);
+    var charge = balance.filter(getExpenseAccount);
 
-    vm.profit = creditorSold(vm.exploitation.profit);
-    vm.charge = debitorSold(vm.exploitation.charge);
+    vm.gridOptions.data = profit.concat(charge);
+    vm.profit = creditorSold(profit);
+    vm.charge = debitorSold(charge);
     vm.globalResult = vm.profit - vm.charge;
   })
   .catch(Notify.handleError);
 
+  // get profit account
+  function getProfitAccount(account) {
```

Comment:
This is clever, but why do you need to filter out `nullBalance` accounts?  Could something like `util.filterFormElements()` work better?